qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -215,6 +202,42 @@ def _on_config_changed(self, option):
             self.maybe_hide()
         elif option == 'statusbar.padding':
             self._set_hbox_padding()
+        elif option == 'statusbar.widgets':
+            self._draw_widgets()
+
+    def _draw_widgets(self):
+        """Draw statusbar widgets."""
+        # Start with widgets hidden and show them when needed
+        for widget in [self.url, self.percentage,
+                       self.backforward, self.tabindex,
+                       self.keystring, self.prog]:
+            widget.hide()
+            self._hbox.removeWidget(widget)
+
+        # Read the list and set widgets accordingly
+        for segment in config.val.statusbar.widgets:
+            if segment == 'url':
+                self._hbox.addWidget(self.url)
+                self.url.show()
+            elif segment == 'scroll':
+                self._hbox.addWidget(self.percentage)
+                self.percentage.show()
+            elif segment == 'scroll_raw':
+                self._hbox.addWidget(self.percentage)
+                self.percentage.raw = True
+                self.percentage.show()
+            elif segment == 'history':
+                self._hbox.addWidget(self.backforward)
+                self.backforward.enabled = True
```

Comment:
When being on a page where the history widget is shown and changing the setting, it's now hidden because it gets added as hidden above and never shown. I guess the best thing to do here would be `self.backforward.on_tab_changed(self._current_tab())` or so to make sure it always has the right state. The same applies to the `progress` widget as well (though it's probably not as visible there as it's only shown for a short time normally).

Reply:
It seems to work.